### PR TITLE
feat: add `docProcessor` to generate docs for each components

### LINKED ISSUE #30: Update build:docs with next stable version of auro-library
FYI, `build:docs` does not do anything due to Auro-library's bug on `^2.9.0` and `2.8.0` has another bug to process `build:docs`. To fix this, we should wait for new stable version of auro-libarary. We can move on and address `build:docs` issue on a separate ticket.

_Originally posted by @sun-mota in https://github.com/AlaskaAirlines/auro-form/issues/16#issuecomment-2460144315_

### PR DESCRIPTION
**Pull request by sun-mota**
# Alaska Airlines Pull Request

closes #30 

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I have performed a self-review of my own update.**

## Summary by Sourcery

Introduce `docProcessor` to automate the generation of documentation for each component, ensuring consistency and up-to-date information across the project. Update file paths in documentation to reflect the new structure.

New Features:
- Introduce `docProcessor` to generate documentation for each component.

Enhancements:
- Update file paths in documentation to reflect new structure.

Documentation:
- Generate new documentation for components using `docProcessor`, ensuring each component has up-to-date API and usage examples.

### CODE REVIEW

**Review by rmenner — APPROVED** [review]
Running locally works. I don't see anything that stands out to me. On a side note. Please resolve all the AI comments.

File: components/input/package.json [hunk 1]
```diff
@@ -54,10 +54,11 @@
     "access": "public"
   },
   "scripts": {
-    "build": "run-s build:version build:sass bundler types",
+    "build": "run-s build:version build:sass bundler build:docs types",
     "build:version": "node ./src/scripts/version.mjs",
     "build:sass": "sass --no-source-map --load-path=../../node_modules \"./src/styles/:./src/styles/\"",
     "build:watch": "nodemon -e scss,js --watch src --ignore \"*.css\" --ignore \"*-css.js\" --exec \"npm run build\"",
+    "build:docs": "wca analyze 'scripts/wca/*.js' --outFiles docs/api.md; wca 'scripts/wca/*' --outFiles docs/api.md; node ../../packages/build-tools/src/docProcessor.mjs --component input",
```
[the remark this answers is not in the document]
Comment (chrisfalaska) [reply components/input/package.json]: @sun-mota This seems like a good suggestion.

File: packages/build-tools/src/docProcessor.mjs [hunk 2]
```diff
@@ -0,0 +1,104 @@
+import { Logger } from "@aurodesignsystem/auro-library/scripts/utils/logger.mjs";
```
Comment (chrisfalaska) [inline packages/build-tools/src/docProcessor.mjs:1]: You'll also want to add an `export` reference in `packages/build-tools/package.json`:

```
"exports": {
    ".": "./src/index.js",
    "./formVersionWriter": "./src/formVersionWriter.mjs",
    "./docProcessor": "./src/docProcessor.mjs"
  }
```

**Review by chrisfalaska — CHANGES_REQUESTED** [review]

File: components/input/demo/api.md [hunk 3]
```diff
@@ -14,98 +14,101 @@
...
-<!-- AURO-GENERATED-CONTENT:START (FILE:src=./../api.md) -->
-<!-- The below content is automatically added from ./../api.md -->
-
+</style>
+<!-- AURO-GENERATED-CONTENT:START (FILE:src=../docs/api.md) -->
+<!-- The below content is automatically added from ../docs/api.md -->
+
 # auro-input
 
+Generate unique names for dependency components.
```
[the remark this answers is not in the document]
Comment (sun-mota) [reply components/input/demo/api.md:23]: will leave as is as there is an original md in auro-input

**Review by chrisfalaska — APPROVED** [review]

**Pull request merged**